skada/metrics.py

```diff
@@ -564,25 +562,20 @@ def _score(self, estimator, X, y, sample_domain=None):
         else:
             y_type = _find_y_type(y[source_idx])
 
+            backward_y = y.copy()
+            backward_y[~source_idx] = y_pred_target
+
             if y_type == Y_Type.DISCRETE:
                 # We need to re-encode the target labels
                 # since some estimator like XGBoost
                 # only supports labels in [0, num_classes-1]
                 # and y_pred_target may not be in this range
                 le = LabelEncoder()
-                le.fit(y_pred_target)
-                y_pred_target = le.transform(y_pred_target)
+                le.fit(backward_y)
+                backward_y = le.transform(backward_y)
 
             backward_sample_domain = -sample_domain
 
-            backward_y = np.zeros_like(y)
-            backward_y[~source_idx] = y_pred_target
-
-            if y_type == Y_Type.DISCRETE:
-                backward_y[source_idx] = _DEFAULT_MASKED_TARGET_CLASSIFICATION_LABEL
-            else:
-                backward_y[source_idx] = _DEFAULT_MASKED_TARGET_REGRESSION_LABEL
-
```

Comment:
I'm not sure that these lines need to be removed. You still need to mask the part of the input that is not considered to be targets, right? 🤔

Reply:
Well every DA methods implemented in skada are supposed to not using the target labels, so feeding them to the methods wont change anything. And with the circular scorer, when we train on the target domain we need to unmasked labels (or it breaks).

I know that usually we mask them, to be sure that we don't have any data leak, bu here I don't see any other solutions

Reply:
I'm not sure what you mean by 'not suppose to use target labels'. The estimator that doesn't request `sample_domain` gets input without separation for source/target (as it cannot operate with it), only masked labels are stripped away.